src/Runner.Worker/SnapshotOperationProvider.cs

```diff
@@ -0,0 +1,35 @@
+﻿#nullable enable
+using System.IO;
+using System.Threading.Tasks;
+using GitHub.DistributedTask.Pipelines;
+using GitHub.Runner.Common;
+using Newtonsoft.Json;
+
+namespace GitHub.Runner.Worker;
+
+[ServiceLocator(Default = typeof(SnapshotOperationProvider))]
+public interface ISnapshotOperationProvider : IRunnerService
+{
+    Task CreateSnapshotRequestAsync(IExecutionContext executionContext, Snapshot snapshotRequest);
+}
+
+public class SnapshotOperationProvider : RunnerService, ISnapshotOperationProvider
+{
+    public async Task CreateSnapshotRequestAsync(IExecutionContext executionContext, Snapshot snapshotRequest)
+    {
+        var snapshotRequestFilePath = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Root), ".snapshot", "request.json");
+        var snapshotRequestDirectoryPath = Path.GetDirectoryName(snapshotRequestFilePath);
+        if (snapshotRequestDirectoryPath != null)
+        {
+            Directory.CreateDirectory(snapshotRequestDirectoryPath);
+        }
+
+        var snapshotRequestJson = JsonConvert.SerializeObject(snapshotRequest);
```

Comment:
you can use `StringUtil.ConvertToJson()` instead of `JsonConvert.SerializeObject`

Reply:
you can also just use `IOUtil.SaveObject()` to do the josn serialization and write to disk in one line.

Reply:
Thanks for the advice! 

I opted for just using `IOUtil.SaveObject`. Originally I thought it made sense to have the operations separated as I was logging the snapshot object JSON afterwards but I ended up removing that as it seemed redundant. The behaviour should be adequately tested by the L0 tests.